File: client/src/app/modules/quotation/services/service.field.type.service.ts
import { HttpClient, HttpParams } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { AppRestService } from 'src/app/services/appRest.service';
import { ServiceFieldType } from '../../quotation/model/ServiceFieldType';

@Injectable({
  providedIn: 'root'
})
export class ServiceFieldTypeService extends AppRestService<ServiceFieldType> {

  constructor(http: HttpClient) {
    super(http, "quotation");
  }

  getServiceFieldTypes() {
    return this.getList(new HttpParams(), "service-field-types");
  }

  addOrUpdateServiceFieldType(serviceFieldType: ServiceFieldType) {
    return this.addOrUpdate(new HttpParams(), "service-field-type", serviceFieldType, "Enregistré", "Erreur lors de l'enregistrement");
  }
}

File: src/main/java/com/jss/osiris/modules/quotation/service/ServiceFieldTypeService.java
package com.jss.osiris.modules.quotation.service;

import java.util.List;

import com.jss.osiris.modules.quotation.model.ServiceFieldType;

public interface ServiceFieldTypeService {
    public List<ServiceFieldType> getServiceFieldTypes();

    public ServiceFieldType getServiceFieldType(Integer id);
	
	 public ServiceFieldType addOrUpdateServiceFieldType(ServiceFieldType serviceFieldType);
}
